security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -421,24 +444,49 @@ func pushSDS(con *sdsConnection) error {
 	if err = con.stream.Send(response); err != nil {
 		sdsServiceLog.Errorf("Failed to send response for SDS resource %q to proxy connection %q: %v",
 			resourceName, conID, err)
+		sdsMetrics.pushFailurePerConn.With(prometheus.Labels{"resourcePerConn": resourceName + "-" + conID}).Inc()
```

Comment:
If I read it correctly, this cause the huge overhead on telemetry backend due to the cardinality of the connection id is unbound.

If we really want to have per connection information, they should not go to prometheus metrics, just print plaintext somewhere, maybe controlz.

Reply:
Yes, see https://github.com/istio/istio/issues/14642

Reply:
@incfly @howardjohn I am not clear if it is possible to print plaintext only at controlz, could you show me an example for doing that? If ctrlz does not support, maybe we need to add an extra endpoint to show this.

Reply:
I am going to add counters without using labels, and move all the labeled metrics into a separate page.

Reply:
@douglas-reid showed me how to drop the metrics with labels so that we can monitor at /metrics but they are not sent to Prometheus. This requires some Prometheus config change.

Reply:
I'm still not sure its useful to add one per connection. When would a user use this to debug?

Reply:
per offline discussion, these metrics per connection help to debug workload SDS issues. The connection ID has a prefix of proxy ID, so that the metrics show which proxy has received SDS push for what SDS resources. If queries to some services respond with 5xx, we could check the metrics to find out whether citadel agent has pushed key/cert to the sidecars.